### DISCUSSION TITLE
Is there any example to multi level authentication?

### DISCUSSION THREAD

**Discussion by rimzzlabs** [supabase/supabase#9260] [Questions]
I have this diagram where my app has 3 different type of users, like for example, I have:

- Admin ( has everything, can create Lecturer or Student )
- Lecturer ( has permission to edit course for current or upcoming semester )
- Student ( has access to what courses they can choose )

Is there any example about this multi level role? I'm sorry if I have misspelled the terms above, thank you

**Comment by GaryAustin1** [comment]
This covers one approach to handling special roles.
https://github.com/supabase-community/supabase-custom-claims
Basically you would use RLS to limit what the roles can do, and have a user roles table, or something like the link using the app metadata, to hold the roles.
